exchange/utils.go

```diff
@@ -761,10 +761,11 @@ func setUserExtWithCopy(request *openrtb2.BidRequest, userExtJSON json.RawMessag
 
 // resolveBidder returns the known BidderName associated with bidder, if bidder is an alias. If it's not an alias, the bidder is returned.
 func resolveBidder(bidder string, aliases map[string]string) openrtb_ext.BidderName {
+	normalisedBidderName, _ := openrtb_ext.NormalizeBidderName(bidder)
 	if coreBidder, ok := aliases[bidder]; ok {
 		return openrtb_ext.BidderName(coreBidder)
 	}
-	return openrtb_ext.BidderName(bidder)
+	return normalisedBidderName
```

Comment:
Couple of points:
- This function will send empty name in case `normalisedBidderName` on line 764 is empty and their is no corresponding `aliases`
- Will this handle case where alias has senstitive bidder name.  As per code, `Appnexus` will be  returned from line 767 but expected bidder name is `appnexus`

   ```
   imp:[
    {
         ...
         ext: {
           ...
           bidder: {
                test1: {
                  ...
                }
            }
         }
       }
   ],
    ext: {
      alias: {
          test1:  Appnexus // Appnexus here is senstitive bidder name
      }
   }
  ```


could restructure function as follow:

```
func resolveBidder(bidder string, aliases map[string]string) openrtb_ext.BidderName {
	var bidderName openrtb_ext.BidderName
	if coreBidder, ok := aliases[bidder]; ok {
		bidderName = openrtb_ext.BidderName(coreBidder) // Appnexus
		biddeName = openrtb_ext.NormalizeBidderName(bidderName) // Appnexus -> appnexus
	} else {
		bidderName, _ = openrtb_ext.NormalizeBidderName(bidder)
	}
	

	if bidderName == "" {
		// return bidder recieved from as input
		// this is to avoid returning empty bidder name incase above code is not able to resolve bidderName
		return openrtb_ext.BidderName(bidder)
	}
	return bidderName
}
```

Reply:
thanks Onkar! Please find my comments inline - 

> * This function will send empty name in case `normalisedBidderName` on line 764 is empty and their is no corresponding `aliases`

This is actually a part of validation. At this point in code, we do not expect the normalisedBidderName to be empty since this is being validated before making the call to resolveBidder function. In case, where we get empty CoreBidder, we are already returning [the error here](https://github.com/prebid/prebid-server/blob/c5b6e39c11ff19502e7e1b38627ea2bacc6ae302/endpoints/openrtb2/auction.go#L1528-L1529) since `validateImpExt` is being called up before resolveBidder function. 

> * Will this handle case where alias has senstitive bidder name.  As per code, `Appnexus` will be  returned from line 767 but expected bidder name is `appnexus`

Yes, This will handle the case where alias has an case sensitive bidder name. While validating the aliases, we are checking if alias has coreBidder name entry and if yes then alias map is being updated with the normalised bidder name. Here is the [specific code that checks that.](https://github.com/prebid/prebid-server/blob/c5b6e39c11ff19502e7e1b38627ea2bacc6ae302/endpoints/openrtb2/auction.go#L1577-L1578)